components/ProposalEditor.tsx

```diff
@@ -236,6 +240,65 @@ export default function ProposalEditor({
     dispatch(removeMessageAction)
   }
 
+  function handleJsonChanged(json: any) {
+    const updateFromJsonAction: ProposalUpdateFromMessage = {
+      type: 'updateFromMessage',
+      message: json,
+    }
+    setEditProposalJson(false)
+    dispatch(updateFromJsonAction)
+  }
+
+  let mainEditor
+  if (editProposalJson) {
+    mainEditor = (
+      <RawEditor
+        json={messageForProposal(proposal)}
+        onChange={handleJsonChanged}
+      ></RawEditor>
+    )
+  } else {
+    mainEditor = (
```

Comment:
i still feel like this should get broken out into a separate component. let's leave it here for now and continue iterating, just wanted to make a note